test/engine/SpatialJoinAlgorithmsTest.cpp

```diff
@@ -1679,4 +1683,52 @@ TEST(SpatialJoin, mixedDataSet) {
 
 }  // namespace boundingBox
 
+namespace runtimeParameters {
+
+using Loc = ad_utility::source_location;
+
+// Helper to compute a `SpatialJoin` using `libspatialjoin` and check if the
+// used number of threads is the expected.
+void testNumberOfThreads(size_t runtimeParamNumThreads,
+                         size_t expectedNumberOfThreads,
+                         Loc sourceLocation = Loc::current()) {
+  auto cleanUp = setRuntimeParameterForTest<"spatial-join-max-threads">(
+      runtimeParamNumThreads);
+  auto l = generateLocationTrace(sourceLocation);
+  auto qec = buildMixedAreaPointQEC();
+  auto leftChild =
+      buildIndexScan(qec, {"?obj1", std::string{"<asWKT>"}, "?geo1"});
+  auto rightChild =
+      buildIndexScan(qec, {"?obj2", std::string{"<asWKT>"}, "?geo2"});
+  SpatialJoinConfiguration config{
+      LibSpatialJoinConfig{SpatialJoinType::INTERSECTS}, Variable{"?geo1"},
+      Variable{"?geo2"}};
+  config.algo_ = SpatialJoinAlgorithm::LIBSPATIALJOIN;
+  std::shared_ptr<QueryExecutionTree> spatialJoinOperation =
+      ad_utility::makeExecutionTree<SpatialJoin>(qec, config, leftChild,
+                                                 rightChild);
+  std::shared_ptr<Operation> op = spatialJoinOperation->getRootOperation();
+  SpatialJoin* spatialJoin = static_cast<SpatialJoin*>(op.get());
```

Comment:
```suggestion
  auto spatialJoin = std::dynamic_pointer_cast<SpatialJoin>(spatialJoinOperation->getRootOperation());
```

Conceptually you don't need raw pointers here, and also you should use dynamic_cast over static_cast in any case.